tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1097,49 +1109,34 @@ void CommandQueue::wrap(DeviceCommand::WrapRegion wrap_region, bool blocking) {
     this->enqueue_command(command, blocking);
 }
 
-void CommandQueue::reset() {
-    // Place the cores into reset since need to update a lot of core info
-    tt::Cluster::instance().assert_risc_reset_at_core(tt_cxy_pair(this->device->id(), this->device->worker_core_from_logical_core(this->issue_queue_reader_core)));
-    tt::Cluster::instance().assert_risc_reset_at_core(tt_cxy_pair(this->device->id(), this->device->worker_core_from_logical_core(this->completion_queue_writer_core)));
-    tt::Cluster::instance().l1_barrier(this->device->id());
-
-    detail::CommandQueueInit(this->device);
-}
-
-void CommandQueue::launch(launch_msg_t& msg) {
-    tt::Cluster::instance().deassert_risc_reset_at_core(tt_cxy_pair(this->device->id(), this->device->worker_core_from_logical_core(this->issue_queue_reader_core)));
-    tt::Cluster::instance().deassert_risc_reset_at_core(tt_cxy_pair(this->device->id(), this->device->worker_core_from_logical_core(this->completion_queue_writer_core)));
-
-    std::unordered_set<CoreCoord> not_done_cores = {
-        this->device->worker_core_from_logical_core(this->issue_queue_reader_core),
-        this->device->worker_core_from_logical_core(this->completion_queue_writer_core)};
-
-    llrt::internal_::wait_until_cores_done(this->device->id(), RUN_MSG_INIT, not_done_cores);
+void CommandQueue::restart() {
+    ZoneScopedN("CommandQueue_restart");
+    tt::log_debug(tt::LogDispatch, "EnqueueRestart for channel {}", this->id);
+    EnqueueRestartCommand command(this->id, this->device, this->manager);
+    this->enqueue_command(command, false);
+    this->wait_finish();
```

Comment:
the restart command is non blocking but it still waits for finish?

Reply:
it is blocking, there is no non-blocking version of it, so I didn't provide "blocking" as a parameter.

Reply:
oh yes I see that, wasn't obvious when just looking at the host code. could you add a `set_finish` when assembling the device command or add a comment specifying that it is blocking